src/utils/add-localhost-aliases.ts

```diff
@@ -0,0 +1,63 @@
+import { resolve4 } from "dns/promises";
+import { appendFile } from "fs/promises";
+import { DOCKER_BRIDGE_NETWORK_GATEWAY } from "./get-inputs";
+
+const HOSTS_FILE = "/etc/hosts";
+
+// Adds aliases intended for localhost in `/etc/hosts`, mapped instead
+// to the Docker Network bridge gateway IP address.
+export const addLocalhostAliases = async ({
+  appUrl,
+  localhostAliases,
+}: {
+  appUrl: string;
+  localhostAliases: string | null;
+}): Promise<void> => {
+  const autoDetectedAlias = await autoDetectAppURlAlias({ appUrl });
+
+  if (!localhostAliases && autoDetectedAlias == null) {
+    return;
+  }
+
+  const aliases = localhostAliases
+    ? localhostAliases.split(",").map((alias) => alias.trim())
+    : [];
+
+  const allAliases = [
+    ...aliases,
+    ...(autoDetectedAlias ? [autoDetectedAlias] : []),
+  ];
+
+  const hostsEntries = allAliases
+    .map((alias) => `${DOCKER_BRIDGE_NETWORK_GATEWAY}\t${alias}`)
+    .join("\n");
+
+  await appendFile(HOSTS_FILE, `\n${hostsEntries}`, { encoding: "utf-8" });
+};
+
+// Attempts to detect if `appUrl` is aliased to `localhost`
+const autoDetectAppURlAlias = async ({
+  appUrl,
+}: {
+  appUrl: string;
+}): Promise<string | null> => {
+  if (!appUrl) {
+    return null;
+  }
+  try {
+    const url = new URL(appUrl);
+    const ipAddresses = await resolve4(url.hostname).catch(() => {
+      return [];
```

Comment:
```suggestion
      return null;
```

Reply:
I want `ipAddresses` to be of type `string[]`.

Reply:
ah yes! makes sense 🤦‍♂️